container_api/src/config.rs

```diff
@@ -9,17 +10,85 @@ use std::{convert::TryFrom, fs::File, io::prelude::*};
 
 #[derive(Deserialize)]
 pub struct Configuration {
-    agents: Option<Vec<AgentConfiguration>>,
-    dnas: Option<Vec<DNAConfiguration>>,
-    instances: Option<Vec<InstanceConfiguration>>,
-    interfaces: Option<Vec<InterfaceConfiguration>>,
-    bridges: Option<Vec<Bridge>>,
+    pub agents: Option<Vec<AgentConfiguration>>,
+    pub dnas: Option<Vec<DNAConfiguration>>,
+    pub instances: Option<Vec<InstanceConfiguration>>,
+    pub interfaces: Option<Vec<InterfaceConfiguration>>,
+    pub bridges: Option<Vec<Bridge>>,
+}
+
+impl Configuration {
+    pub fn check_consistency(&self) -> Result<(), String> {
+        if self.instances.is_none() {
+            return Err("No instance found".to_string());
+        }
+        for ref instance in self.instances.as_ref().unwrap().iter() {
+            self.agent_by_id(&instance.agent).is_some().ok_or_else(|| {
+                format!(
+                    "Agent configuration {} not found, mentioned in instance {}",
+                    instance.agent, instance.id
+                )
+            })?;
+            self.dna_by_id(&instance.dna).is_some().ok_or_else(|| {
+                format!(
+                    "DNA configuration \"{}\" not found, mentioned in instance \"{}\"",
+                    instance.dna, instance.id
+                )
+            })?;
+        }
+        if self.interfaces.is_some() {
+            for ref interface in self.interfaces.as_ref().unwrap().iter() {
+                for ref instance in interface.instances.iter() {
+                    self.instance_by_id(&instance.id).is_some().ok_or_else(|| {
+                        format!(
+                            "Instance configuration \"{}\" not found, mentioned in interface",
+                            instance.id
+                        )
+                    })?;
+                }
+            }
+        }
```

Comment:
Looks like you're building `Result<(), String>` values but aren't using them?

Reply:
Am I? Not sure what you mean. `check_consistency()` and the error message is used here: https://github.com/holochain/holochain-rust/pull/605/files#diff-927e383d6b6393e9f319091b2e4fd147R54

Reply:
Ahhh, sorry I missed the `?` on L47.